### PR TITLE
Create new SQA training material

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

**Comment by aeslaughter** [comment]
There are two presentations here that need some eyes:
https://mooseframework.inl.gov/docs/PRs/18919/site/sqa/index.html

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+### PLN-4005: Static Documentation
+
+- +Software Quality Assurance Plan (SQAP; PLN-4005)+: Defines [!ac](SQA)
+  processes implemented including Configuration Management Plan (CMP) and Verification
+  and Validation Plan (VVP)\*.
+- +Safety-software Determination (SSD)+: [!ac](INL) form to declare if the application is safety
+  software
+- +Quality-level Determination (QLD)+: [!ac](INL) form to declare the safety software level
+- +[!ac](EA) Entry+: [!ac](INL) from that lists [!ac](SQA) documentation for application
```
Comment (loganharbour) [inline modules/doc/training/content/sqa/training/ccb/pln-4005.md:56]: No form #s for these?

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+!---
+
+## PLN-4005: Reviews
+
+- +Static Documentation Review+: Changes to the static documentation follow the review procedure
+  native to the INL system (SQA Lead should handle these)
+
+- +Design Verification Review+: "A review shall be conducted by an independent reviewer. The reviewer
+  may rely on assistance from any individual, but the final approval or disapproval is the sole
+  responsibility of the independent reviewer. All comments are retained in the GitHub/GitLab system.
```
Comment (loganharbour) [inline modules/doc/training/content/sqa/training/ccb/pln-4005.md:100]: Missing end quote

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+- +Release Review+: A release requires an additional release review...to ensure compliance with the
+  approved software requirements. Test results from CIVET are considered part of the review.
```
Comment (loganharbour) [inline modules/doc/training/content/sqa/training/ccb/pln-4005.md:103]: The ellipsis looks weird here - should this be a quote?

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+!---
+
+## PLN-4005: Standards, Practices, Conventions, and Metrics
+
+MOOSE and MOOSE-based applications must follow a coding standard that is defined in the
+Software Coding Standard (SCS) document.
+
+Software development of MOOSE and MOOSE-based applications relies on an continuous agile integration
+process using test-driven development through pull/merge requests...
```
Comment (loganharbour) [inline modules/doc/training/content/sqa/training/ccb/pln-4005.md:160]: ellipsis!? why

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+
+!---
+
+## PLN-4005: Support Software
+
+- +GitHub/GitLab+: repository service, change requests, and issue tracking
+- +System Software+: identified by the project lead, documented within the test results, and is
```
Comment (loganharbour) [inline modules/doc/training/content/sqa/training/ccb/pln-4005.md:167]:
```suggestion
- +GitHub/GitLab+: Repository service, change requests, and issue tracking
- +System Software+: Identified by the project lead, documented within the test results, and is
```

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+## PLN-4005: Software Test Execution
+
+Five steps must exist between a change request and a release.
+
+1. +Change Request Testing+: Pull/Merge Request testing
+1. +Code Review+: Independent review of changes by [!ac](CCB) member
+1. +Development Branch Testing+: ensure changes work together
+1. +Documentation+: build current website/docs
+1. +Release+: final review and release by Project Lead (0INL1862)
```
Comment (loganharbour) [inline modules/doc/training/content/sqa/training/ccb/pln-4005.md:150]:
```suggestion
1. +Development Branch Testing+: Ensure the changes work together
1. +Documentation+: Build current website/documentation
1. +Release+: Final review and release by Project Lead (0INL1862)
```

**Review by loganharbour — CHANGES_REQUESTED** [review]
I looked through both presentations. all minor changes

**Comment by cticenhour** [comment]
Stand by @aeslaughter, I've got a list of review comments as well that may
overlap with some of this.

File: modules/doc/training/content/sqa/training/lead/sqa_release.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+Author: moosetest <bounces@inl.gov>
+Date:   Wed Sep 29 11:22:23 2021 -0600
+
+    Merge commit '4812111857205877df7bb943958f9cd1f32b0d27'
+```
+
+The hashes in this message can be used to obtain results from CIVET, as shown in the links below.
+The downloaded tag.gz files should be committed in a "release" directory at the top level of the
+repository. The files should be renamed to follow the pattern "results_12345.tar.gz", where the
+number is the job number.
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_release.md:57]: For future training clarification, perhaps "job number" should be described in a little more detail?

File: modules/doc/training/content/sqa/training/lead/sqa_release.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+[`https://civet.inl.gov/sha_events/idaholab/moose/b657ae26aae14fbc91ad2651a364cfad80c2b9ac/`](https://civet.inl.gov/sha_events/idaholab/moose/b657ae26aae14fbc91ad2651a364cfad80c2b9ac/)\\
+[`https://civet.inl.gov/sha_events/idaholab/moose/4812111857205877df7bb943958f9cd1f32b0d27/`](https://civet.inl.gov/sha_events/idaholab/moose/4812111857205877df7bb943958f9cd1f32b0d27/)
+
+!---
+
+## Perform Release Review
+
+PLN-4005 details the necessary steps for performing the release which includes performing a
+final review of MOOSE or MOOSE-based application. This review +must+ be performed by the Project Lead
+and is intended to ensure that all requirements are satisfied and the all [!ac](SQA) documentation
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_release.md:69]:
```suggestion
and is intended to ensure that all requirements are satisfied and that all [!ac](SQA) documentation
```

File: modules/doc/training/content/sqa/training/lead/sqa_release.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+
+!---
+
+## Create Release Log
+
+The "release log" provides evidence that the final review of software was conducted. This log is
+a text file ("RELEASE_LOG.md") that should be added to the "release" directory of the repository.
+
+!alert note
+Please refer to the current version of PLN-4005 to determine what is necessary, do not
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_release.md:80]:
```suggestion
Please refer to the current version of PLN-4005 to determine what is necessary. Do not
```

File: modules/doc/training/content/sqa/training/lead/sqa_release.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+rely on a log from a previous release. The necessary components of the log might have been
+altered since the previous release.
+
+!---
+
+## Update MooseDocs config.yml
+
+To ensure that the results displayed on the generate website are correct and pulled from the
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_release.md:88]:
```suggestion
To ensure that the results displayed on the generated website are correct and pulled from the
```

File: modules/doc/training/content/sqa/training/lead/sqa_release.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+                 repo: idaholab/moose
+                 location: ${MOOSE_DIR}
+```
+
+!---
+
+## Commit Release
+
+After the testing results are downloaded, the release log created, and the "config.yml" are
+created the additions must be committed to the repository. It is suggested that the following
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_release.md:123]:
```suggestion
After the testing results are downloaded, the release log created, and the "config.yml" is
created, the additions must be committed to the repository. It is suggested that the following
```

File: modules/doc/training/content/sqa/training/lead/sqa_release.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+!alert note
+Depending on your configuration the main repository may be named something other than "upstream."
+
+!---
+
+## Upload Release Website (optional)
+
+Although not required by PLN-4005 the website for the release should be uploaded for consumption
+by users of MOOSE or MOOSE-based application. Please contact the MOOSE developers for assistance
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_release.md:151]:
```suggestion
by users of MOOSE or the MOOSE-based application. Please contact the MOOSE developers for assistance
```

File: modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+```
+cd doc
+./moosedocs.py build --serve
+```
+
+!---
+
+## Update Content
+
+To allow for the design documentation to be complete, the content [!ac](MOOSE) framework and
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md:25]:
```suggestion
To allow for the design documentation to be complete, the content of the [!ac](MOOSE) framework and
```

File: modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+        unregister:
+            framework: !include framework/doc/unregister.yml
+        includes:
+            - ${ROOT_DIR}/include
+            - ${MOOSE_DIR}/framework/include
+            - ${MOOSE_DIR}/modules/heat_conduction/include
+```
+
+!alert note
+This is not necessary for NQA-1 compliance, but it is could practice for creating complete documentation.
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md:62]:
```suggestion
This is not necessary for NQA-1 compliance, but it is good practice for creating complete documentation.
```

File: modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+!---
+
+## SQA Reports
+
+To aid in monitoring the status of [!ac](SQA) documentation a command-line reporting system exists.
+To use this system a report configuration file ("sqa_reports.yml") must be created in the "doc"
+directory of the MOOSE-based application.
+
+This file should contain three sections: "Applications", "Documents", and "Requirements". In the
+following slides the configuration from the "blackbear" application are presented. The file
```
Comment (cticenhour) [inline modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md:125]:
```suggestion
following slides the configuration from the "blackbear" application is presented. The file
```

**Review by cticenhour — DISMISSED** [review]

**Comment by cticenhour** [comment]
All relatively minor changes from me - mostly grammar and punctuation suggestions and a few comments.

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md (hunk 3 above)
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/ccb/pln-4005.md:103]: Yes, but it doesn't matter.

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md (hunk 4 above)
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/ccb/pln-4005.md:160]: It is a partial quote, I fixed that.

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/ccb/pln-4005.md:167]: Yes

File: modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md [hunk 16]
```diff
@@ -0,0 +1,284 @@
...
+        repos:
+            default: https://github.com/idaholab/blackbear
+            moose: https://github.com/idaholab/moose
+        categories:
+            framework: !include ${MOOSE_DIR}/framework/doc/sqa_framework.yml
+            heat_conduction: !include ${MOOSE_DIR}/modules/heat_conduction/doc/sqa_heat_conduction.yml
+            blackbear: !include ${ROOT_DIR}/doc/sqa_blackbear.yml
+```
+
+!alert note
```
[the remark this answers is not in the document]
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/lead/sqa_moosedocs.md:232]: fixed it

File: modules/doc/training/content/sqa/training/lead/sqa_release.md (hunk 7 above)
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/lead/sqa_release.md:57]: I am going to remove the need to rename the files

File: modules/doc/training/content/sqa/training/lead/sqa_civet.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+abort_on_failure = False
+
+# Add testing steps here
+
+[coverage: report]
+script = scripts/coverage_extract_and_store.sh
+COVERAGE_TAG = main
+```
+
+!alert note
```
[the remark this answers is not in the document]
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/lead/sqa_civet.md:68]: I fixed this up

File: modules/doc/training/content/sqa/training/ccb/pln-4005.md (hunk 1 above)
Comment (aeslaughter) [reply modules/doc/training/content/sqa/training/ccb/pln-4005.md]: Not that I can find, the are online sites

**Review by loganharbour — APPROVED** [review]

**Pull request merged**